cmd/maya-apiserver/cstor-operator/cspc/storagepool_create.go

```diff
@@ -0,0 +1,362 @@
+	}
+}
+
+func getPodLabels(csp *apis.NewTestCStorPool) map[string]string {
+	return getPodLabels(csp)
+}
+
+func getPodAnnotations() map[string]string {
+	return map[string]string{
+		"openebs.io/monitoring": "pool_exporter_prometheus",
+		"prometheus.io/path":    "/metrics",
+		"prometheus.io/port":    "9500",
+		"prometheus.io/scrape":  "true",
+	}
+}
+
+func getDeployMatchLabels(csp *apis.NewTestCStorPool) map[string]string {
+	return map[string]string{
+		"app": "cstor-pool",
+	}
+}
+
+// getVolumeTargetImage returns Volume target image
+// retrieves the value of the environment variable named
+// by the key.
+func getPoolMgmtImage() string {
+	image, present := os.LookupEnv("OPENEBS_IO_CSTOR_POOL_MGMT_IMAGE")
+	if !present {
+		image = "openebs/cstor-pool-mgmt:latest"
+	}
+	return image
+}
+
+// getVolumeTargetImage returns Volume target image
+// retrieves the value of the environment variable named
+// by the key.
+func getPoolImage() string {
+	image, present := os.LookupEnv("OPENEBS_IO_CSTOR_POOL_IMAGE")
+	if !present {
+		image = "openebs/cstor-pool:latest"
+	}
+	return image
+}
+
+// getVolumeTargetImage returns Volume target image
+// retrieves the value of the environment variable named
+// by the key.
+func getMayaExporterImage() string {
+	image, present := os.LookupEnv("OPENEBS_IO_CSTOR_POOL_EXPORTER_IMAGE")
+	if !present {
+		image = "openebs/m-exporter:latest"
+	}
+	return image
+}
+
+func getContainerPort(port ...int32) []corev1.ContainerPort {
+	var containerPorts []corev1.ContainerPort
+	for _, p := range port {
+		containerPorts = append(containerPorts, corev1.ContainerPort{ContainerPort: p, Protocol: "TCP"})
+	}
+	return containerPorts
+}
+
+func getPoolMgmtMounts() []corev1.VolumeMount {
+	return append(
+		defaultPoolMgmtMounts,
+		corev1.VolumeMount{
+			Name:      "sparse",
+			MountPath: getSparseDirPath(),
+		},
+	)
+}
+
+func getSparseDirPath() string {
+	dir, present := os.LookupEnv("OPENEBS_IO_CSTOR_POOL_SPARSE_DIR")
+	if !present {
+		dir = "/var/openebs/sparse"
+	}
+	return dir
+}
+
+func getPoolMgmtEnv(csp *apis.NewTestCStorPool) []corev1.EnvVar {
+	var env []corev1.EnvVar
+	return append(
+		env,
+		corev1.EnvVar{
+			Name:  "OPENEBS_IO_CSTOR_ID",
+			Value: string(csp.GetUID()),
+		},
+		corev1.EnvVar{
+			Name: "RESYNC_INTERVAL",
+			// TODO : Add tunable
+			Value: "30",
+		},
+		corev1.EnvVar{
+			Name: "POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		},
+		corev1.EnvVar{
+			Name: "NAMESPACE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+	)
+}
+
+func getPoolLivenessProbe() *corev1.Probe {
+	probe := &corev1.Probe{
+		Handler: corev1.Handler{
+			Exec: &corev1.ExecAction{
+				Command: []string{"/bin/sh", "-c", "zfs set io.openebs:livenesstimestap='$(date)' cstor-$OPENEBS_IO_CSTOR_ID"},
```

Comment:
command similar to this in CAST is not working.. so, please verify this once

Reply:
thing that is not working is '$(date)'.. this property value was set to '$(date)'